specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-04-02-preview/managedClusters.json

```diff
@@ -6177,6 +6397,138 @@
     "CapacityReservationGroupID": {
       "description": "Capacity Reservation Group ID for AgentPool to associate",
       "type": "string"
+    },
+    "TrustedAccessRoleRule": {
+      "type": "object",
+      "properties": {
+        "verbs": {
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "type": "string"
+          }
+        },
+        "apiGroups": {
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "type": "string"
+          }
+        },
+        "resources": {
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "type": "string"
+          }
+        },
+        "resourceNames": {
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "type": "string"
+          }
+        },
+        "nonResourceURLs": {
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "TrustedAccessRole": {
+      "type": "object",
+      "properties": {
+        "sourceResourceType": {
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "type": "string",
+          "readOnly": true
+        },
+        "rules": {
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/TrustedAccessRoleRule"
+          },
+          "x-ms-identifiers": []
+        }
+      }
+    },
+    "TrustedAccessRoleListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/TrustedAccessRole"
+          },
+          "x-ms-identifiers": [
+            "sourceResourceType",
+            "name"
+          ]
+        }
+      }
+    },
+    "TrustedAccessRoleBindingProperties": {
+      "type": "object",
+      "required": [
+        "sourceResourceId",
+        "roles"
+      ],
+      "properties": {
+        "provisioningState": {
+          "type": "string",
+          "readOnly": true
+        },
+        "sourceResourceId": {
+          "type": "string"
+        },
+        "roles": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "TrustedAccessRoleBinding": {
+      "type": "object",
+      "required": [
+        "properties"
+      ],
+      "allOf": [
+        {
+          "$ref": "#/definitions/SubResource"
+        }
+      ],
+      "properties": {
+        "systemData": {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/systemData",
+          "readOnly": true,
+          "description": "The system metadata relating to this resource."
+        },
```

Comment:
I would appreciate it if you could replace the custom `SubResource` with `Resource` from common types (e.g., `"$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/Resource"`), which includes the properties in `SubResource` and the `systemData`. We may remove the custom `SubResource` in the future.

Reply:
sure, updated. Thanks for reminder.